### PR TITLE
[MM-20396] Convert copy_url_context_menu to TypeScript

### PR DESCRIPTION
**Pull request by LauSam09**
#### Summary
This pull request converts the `copy_url_context_menu` component and tests to TypeScript.
The original staticProp for `siteURL` was a non-optional prop - however as it is mapped to a type of `Partial<ClientConfig>` in redux, I had to make it optional in order to prevent compilation errors.

#### Ticket Link
Fixes mattermost/mattermost-server#16710
https://mattermost.atlassian.net/browse/MM-20396

### CODE REVIEW

File: components/copy_url_context_menu/copy_url_context_menu.tsx [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
 
-        actions: PropTypes.shape({
-            copyToClipboard: PropTypes.func.isRequired,
-        }),
+    actions: {
+        copyToClipboard: (link: string) => void;
     };
+}
 
+export default class CopyUrlContextMenu extends React.PureComponent<Props> {
```
Comment (AGMETEOR) [inline components/copy_url_context_menu/copy_url_context_menu.tsx:32]: Hey @LauSam09 . Thanks a lot for this good work. We really appreciate you helping us out. Everything looks good and I just have a small nit/request, Is it possible to also type the return of the render() function? ```render(): JSX.Elelment``` :)

Comment (LauSam09) [reply components/copy_url_context_menu/copy_url_context_menu.tsx:32]: Hello @AGMETEOR, you're welcome! Yeah of course, I'll add them now. I looked at previous PRs and didn't see any examples of people defining return types for functions which seemed weird given that `eslint` was very unhappy about it!

Comment (LauSam09) [reply components/copy_url_context_menu/copy_url_context_menu.tsx:32]: @AGMETEOR I've taken your comment literally and added a return type just for the `render` method - if this was right I'm curious why the other functions shouldn't have return types defined?

**Review by nevyangelova — APPROVED** [review]
Thanks @LauSam09 LGTM!

File: components/copy_url_context_menu/copy_url_context_menu.tsx (hunk 1 above)
Comment (AGMETEOR) [reply components/copy_url_context_menu/copy_url_context_menu.tsx:32]: Sorry if I made some confusion. Atm, the eslint rules don't enforce it. Hopefully, I have answered your question :)

**Review by AGMETEOR — APPROVED** [review]

**Comment by AGMETEOR** [comment]
@hanzei anyone in QA we can request to do a review?

**Review by jgilliam17 — APPROVED** [review]
Thank you @LauSam09 
Tested, looks good to merge.
- Verified copy URL menu and function - as before.
- 
@AGMETEOR Can you please help merge? Thanks 🙂

**Comment by AGMETEOR** [comment]
Thanks @LauSam09

**Pull request merged**